geomstats/_backend/tensorflow/__init__.py

```diff
@@ -294,50 +297,15 @@ def _assignment_single_value_by_sum(x, value, indices, axis=0):
             n_samples, indices, tile_shape, axis, x.dtype)
     else:
         mask = _mask_from_indices(indices, shape(x), x.dtype)
-    x_new = x + value * mask
-    return x_new
-
-
-def assignment_by_sum(x, values, indices, axis=0):
-    """Add values at given indices of an array.
-
-    Parameters
-    ----------
-    x: array-like, shape=[dim]
-        Initial array.
-    values: {float, list(float)}
-        Value or list of values to be assigned.
-    indices: {
-    int, tuple(int), array-like({int, tuple, boolean})
-        Single index or array of indices where values are assigned.
-        If the length of the tuples is shorter than ndim(x) by one, values are
-        assigned to each copy along axis.
-        If indices is a list of booleans and ndim(x) > 1, values are assigned
-        across all dimensions.
-    axis: int, optional
-        Axis along which values are assigned, if vectorized.
+    if mode == 'replace':
+        return x + -x * mask + value * mask
+    if mode == 'sum':
+        return x + value * mask
+    else:
+        raise ValueError('mode must be one of \'replace\' or \'sum\'')
```

Comment:
This line is not tested.

Reply:
True, but how do I test it ? I chose to add the `mode` parameter to switch between cases, keeping it as a string to allow creating new cases later. However, as it is an internal function, specific to a backend, I don't really see how to explicitely test this ValueError case.
Any idea ?

Reply:
Would `assertRaises( ... with gs._assignment_single_value(mode='not_a_mode') ...)` work?

Reply:
Yes, but only in tf backend, as this internal function does not exist for numpy and pytorch

Reply:
Ah ok! `if geomstats.tests.tf_backend():` ... do the test.

Reply:
The only thinkg (I didn't think of that until now), is that since it is a backend function, it would need to be added to the list of function that are imported.
Otherwise, we'd need to add a `from geomstats._backend.tensorflow import _assignment_single_value` ... Sounds a bit ugly.

Reply:
Calling `gs._assignment_single_value` only `if geomstats.tests.tf_backend()` wouldn't do it?

Reply:
It won't find it if it's not explicitely put in the function exported by the backend, sadly

Reply:
Ok! Well, too bad I guess! Let's merge!